### ISSUE TITLE
Graphics issues with apps like tiktok when restarting waydroid

### ISSUE THREAD

**Issue by RaziFalah** [waydroid/waydroid#1002]
Hello there! i have waydroid on my ubuntu machine. it works fine and everything runs as expected. but i do have on issue with tiktok.
everytime i stop Waydroid using the ```waydroid session stop``` command, the app experience really serious graphics problems on the next launch.
not all the app but most videos and by most i mean the whole feed page is just unusable, i will provide a screenshots and a gif to describe the problem:

![image](https://github.com/waydroid/waydroid/assets/92949627/b1eea9b6-b30e-4276-8355-49d1e3bd42ad)

![gif](https://shipshape-fifties.000webhostapp.com//ezgif-4-625410f37a.gif)
If the gif  second didn't show up please see the following url  [https://shipshape-fifties.000webhostapp.com//ezgif-4-625410f37a.gif](https://shipshape-fifties.000webhostapp.com//ezgif-4-625410f37a.gif) 

I would like to note that the problem gets solved when i clear the app data and force stop it. but it's just not practical to do this everytime i open waydroid. also i lose all data in the app since most data tiktok store is at the client end. 

**General information:**
` - Waydroid tools Version [e.g. 1.4.1]`

**Desktop:**
 - OS: Ubuntu 23.04 (Lunar Lobster)
 - GPU: Nvidia gtx 1050 ti lp

Nothing useful in the log file, no error was captured.

**Comment by cniw** [comment]
Open Android **Settings** => **Apps & notifications** => **TikTok** => **Storage & cache** => **Clear storage** and **Clear cache**

**Comment by YasirRWebDesigner** [comment]
facing the same issue

**Comment by RaziFalah** [comment]
> facing the same issue

Yes, I would like to update that this same issue is universal. Meaning I face graphics issues in almost all apps that stream videos. For instance reels in the instagram app unexpectedly stop playing and I had to stop the whole waydroid software and start it again in order to solve this issue. But it is a matter of time before it comes back again.

**Comment by itdimk** [comment]
Facing the same issue.
One more thing: live streams works fine for some reason (even after re-launch)
Normal videos looks like glitchy split screen for me.
And yeah, wipe data fixes problem until next tiktok launch

![1](https://github.com/waydroid/waydroid/assets/24206455/abbcde68-3120-4932-93a6-6def20bbb6b5)
![2](https://github.com/waydroid/waydroid/assets/24206455/26ad0e81-900f-43de-8562-e91577e0e1ab)

**Comment by RaziFalah** [comment]
I think this issue is from anbox not waydroid, I'm not sure if they can fix it here.

**Comment by RaziFalah** [comment]
> Facing the same issue. One more thing: live streams work fine for some reason (even after re-launch) Normal videos looks like glitchy split screen for me. And yeah, wipe data fixes problem until next tiktok launch
> 
> ![1](https://private-user-images.githubusercontent.com/24206455/289962612-abbcde68-3120-4932-93a6-6def20bbb6b5.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0MTExNTAsIm5iZiI6MTcwMjQxMDg1MCwicGF0aCI6Ii8yNDIwNjQ1NS8yODk5NjI2MTItYWJiY2RlNjgtMzEyMC00OTMyLTkzYTYtNmRlZjIwYmJiNmI1LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEyVDE5NTQxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyMzZmNWFjZmMyOTBhOTMzZmVkOTU1MDhlOWQwM2U2ZTU1ZDQ3OWUxZmZkZmM2ZDJlOTk3NmMxMGY1MmNiZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.e7cZDGiCQ9iJaoSMyqmaL_BipOZbXkagTLzq7ONlvYg) ![2](https://private-user-images.githubusercontent.com/24206455/289962618-26ad0e81-900f-43de-8562-e91577e0e1ab.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0MTExNTAsIm5iZiI6MTcwMjQxMDg1MCwicGF0aCI6Ii8yNDIwNjQ1NS8yODk5NjI2MTgtMjZhZDBlODEtOTAwZi00M2RlLTg1NjItZTkxNTc3ZTBlMWFiLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEyVDE5NTQxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI1NmE0ZGQ1MmIzNWU3NGZlNjEzMWQyMmU0N2U5NzA0NjIxNDEwNTY4N2Q3YzQwMTU1YzNhN2ViM2I2MDFjZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.iSokCf0f7kFDZKk_ybkD15DsQhwhaHekZfAAORFntcY)


I think this issue is from anbox not waydroid, since waydroid is built on anbox I'm not sure if they can fix it here.

**Comment by fir3-1ce** [comment]
> Open Android **Settings** => **Apps & notifications** => **TikTok** => **Storage & cache** => **Clear storage** and **Clear cache**

This fixed a video issue I was having. Tiktok is so hard to get a good experience out of if you're privacy-conscious. Now I can't get the audio to work. Unrelated, but I think my VPN causes a lot of problems too

**Comment by fir3-1ce** [comment]
Commenting to say it looks like this issue still persists, only I get a black screen whenever video is supposed to be playing. It seems like it should be easily fixable, because sometimes I see a split second of the video when scrolling.

This issue apparently caused me to lose an account today, as after clearing the storage and cache I couldn't log back in. Tried another account and now the video is working again. Super annoying and hard to work with.

**Comment by bon30** [comment]
You can use an older version of tiktok (ver. 20.0 or lower) works in every android emulator. Unfortunately, you can't view pictures with that version.

**Comment by billbeans** [comment]
I suspect it's a matter of Tiktok detecting the VM and using some kind of throttling. Certain Magisk modules might be useful in testing this, if someone who knows what they're doing can successfully hide Waydroid's VM status

**Comment by keplersj** [comment]
Now that TikTok is back on the Play Store in the United States, I thought I'd try installing it on my desktop. Captured that experience [on Bluesky](https://bsky.app/profile/keplersj.com/post/3liaqituvyc2j) and accidentally recreated this bug,

I can hear audio but cannot see videos for posts. I can see photos and live streams, though. No graphical disruptions other than the black screen.

![Image](https://github.com/user-attachments/assets/aa5eec84-b056-4cf4-b10e-c5dbf22bc0f6)

Running Android 11 (with GApps) with Waydroid version 1.4.3 on Fedora Silverblue 41 and configured with `libhoudini`, `magisk`, and `widevine` configured with [`waydroid_script`](https://github.com/casualsnek/waydroid_script).

**Comment by 867380699** [comment]
您发给我的信件已经收到,谢谢!

**Comment by billbeans** [comment]
@keplersj, I'm almost 100% certain that is intentional on Tiktok's end. Something is triggering their bot/spam detection. Are you using a VPN?

My only recourse so far has been to create a webapp for Tiktok using Linux Mint's [Webapp Manager](https://github.com/linuxmint/webapp-manager), and even then I've ran into many restrictions. I cannot follow anyone on my default VPN connection, but when I connect with a good proxy I can follow again. But even then, following more than 10 or so people in a row causes it to "lock" for an indeterminate amount of time.
